docs(Backup-Restore): Rework for Truetool/restore/migration (#57)
truecharts/website: docs/manual/SCALE Apps/backup-restore.md

---
sidebar_position: 15
---
# Backup, Migrations and Restore

:::caution Best Effort Policy

> This guide has been written with the best efforts of the staff and tested as best possible. We are not responsible if it doesn't work for every scenario or user situation. This guide has been test with TrueNAS SCALE 22.02.4 and once 22.12 is production ready we shall update and modify as necessary.

Reviewer: ```suggestion
This guide has been written with the best efforts of the staff and tested as best possible. We are not responsible if it doesn't work for every scenario or user situation. 
This guide has been thoroughly tested with TrueNAS SCALE 22.02.4.
```

:::

## Requirements

> This guide makes use of our command-line tool, called `TrueTool` and assumes you've already created backups using the BASH version.

Reviewer: ```suggestion
This guide makes use of our command-line tool, called `TrueTool` and assumes you've already created backups using the BASH TrueTool.
```

To revert an existing system, the process is as follows:

1. List your current backups using `truetool`

2. Select option 5 `Restore a Backup`

3. Choose Backup and answer the prompt

![TrueTool-RestoreList](img/TrueTool-Restore-List.png)

Please keep in mind this can take a LONG time.

### Total System restore and Migration to new system

>Sometimes you either need to wipe your Pool, Migrate to a new Pool or restore a system completely.
With the steps below, this is all very-much-possible.

> :::caution Do not Choose an apps pool
> 
> Do not choose an apps pool yet, or do ANYTHING with apps until step 3
> 
> :::

Reviewer: ```suggestion
:::caution Do not Choose an Apps pool

Do not choose an apps pool yet, or do ANYTHING with apps until step 3

:::

:::caution Do not restore SCALE-config yet

Please do not restore SCALE configuration from backup-file, before Apps pool replication is finished.
:::

```

Please keep in mind this can take a LONG time.

### Total System restore and Migration to new system

>Sometimes you either need to wipe your Pool, Migrate to a new Pool or restore a system completely.
With the steps below, this is all very-much-possible.

:::caution Do not Choose an Apps pool

Do not choose an apps pool yet, or do ANYTHING with apps until step 3

:::

:::caution Do not restore SCALE-config yet

Please do not restore SCALE configuration from backup-file, before Apps pool replication is finished.
:::

1. _Optional_: When the SCALE system itself is also wiped, ensure to restore it using a SCALE config export **first**.

2. Using ZFS replication, move back the previously backed-up `ix-applications` dataset to the disk that will contain the future Apps Pool. This was covered in the [Exporting Backups](#exporting-backups) section.

3. Once the ZFS replication is complete, on the new or migrated system navigate to the __Apps__ tab in the Truenas Scale GUI. When prompted to select a pool, select the pool containing the `ix-applications` dataset.

4. Run the following command in your Truenas Scale shell to fix known issues with PVC storage.

>
```bash

> zfs set mountpoint=legacy $(zfs list -o name | grep ix-applications | grep pvc) 

Reviewer: ```suggestion
zfs set mountpoint=legacy "$(zfs list -t filesystem -r "$(cli -c 'app kubernetes config' | grep -E "pool\s\|" | awk -F '|' '{print $3}' | tr -d " \t\n\r")" -o name -H | grep "volumes/pvc")" 
```
